internal/provider/resource_hvn_peering_connection.go

```diff
@@ -0,0 +1,289 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/client/network_service"
+	networkmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func resourceHvnPeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description:   "The HVN peering connection resource allows you to manage a network peering connection between HVNs.",
+		CreateContext: resourceHvnPeeringConnectionCreate,
+		ReadContext:   resourceHvnPeeringConnectionRead,
+		DeleteContext: resourceHvnPeeringConnectionDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &peeringDefaultTimeout,
+			Create:  &peeringCreateTimeout,
+			Delete:  &peeringDeleteTimeout,
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceHvnPeeringConnectionImport,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"peering_id": {
+				Description:      "The ID of the network peering.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"hvn_1": {
+				Description: "The unique URL of one of the HVNs being peered.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"hvn_2": {
+				Description: "The unique URL of one of the HVNs being peered.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			// Computed outputs
+			"organization_id": {
+				Description: "The ID of the HCP organization where the network peering is located. Always matches the HVNs' organization.",
```

Comment:
Let's standardize on `peering connection` in this one:
```suggestion
				Description: "The ID of the HCP organization where the peering connection is located. Always matches the HVNs' organization.",
```